lib/lightstep/tracer/tracer.rb

```diff
@@ -0,0 +1,374 @@
+require 'json'
+require 'concurrent'
```

Comment:
Tracer is way more likely to be used by many threads at once, and it's internal storage arrays were very unsafe. One span is much less likely to be used by multiple threads at once, and the only unsafe part of Span is baggage. I can threadsafe baggage if you'd like, although we talked about moving it to SpanContext so I didn't do it yet.